frontend/src/components/pages/acls/new-acl/ACLDetails.tsx

```diff
@@ -0,0 +1,272 @@
+  sharedConfig: {
+    principal: string;
+    host: string;
+  };
+  rules: Rule[];
+  onUpdateACL: () => void;
+  showMatchingUsers?: boolean;
+  isSimpleView?: boolean; // this prop show SharedConfig, this is used for emmbedded this component on legacy user page
+}
+
+export function ACLDetails({
+  sharedConfig,
+  rules,
+  onUpdateACL,
+  showMatchingUsers = false,
+  isSimpleView = false,
+}: ACLDetailsProps) {
+  const nav = useNavigate();
+
+  const getGoTo = (sc: SharedConfig) => {
+    if (sc.principal.startsWith('User:')) {
+      return `/security/acls/${sc.principal.split(':')[1]}/details`;
+    }
+    if (sc.principal.startsWith('RedpandaRole:')) {
+      return `/security/roles/${sc.principal.split(':')[1]}/details`;
+    }
+    return '';
+  };
+
+  const data = {
+    sharedConfig,
+    rules,
+    principalType: sharedConfig.principal.split(':')[0],
+    hostType: sharedConfig.host === '*' ? 'Allow all hosts' : 'Specific host',
+  };
+
+  return (
+    <div>
+      {/* Header */}
+      <div className={`${isSimpleView ? 'hidden' : ''}`}>
+        <div className=" py-4">
+          <div className=" mx-auto">
+            <div className="flex items-center justify-between">
+              <div className="flex items-center gap-4">
+                <div>
+                  <p className="text-sm text-gray-600">View the created ACL configuration</p>
+                </div>
+              </div>
+              <Button
+                className="bg-gray-900 hover:bg-gray-800 text-white"
+                onClick={onUpdateACL}
+                data-testid="update-acl-button"
+              >
+                <Edit className="w-4 h-4 mr-2" />
+                Edit
+              </Button>
+            </div>
+          </div>
+        </div>
+      </div>
+
+      {/* Main Content */}
+      <main>
+        <div className=" mx-auto">
+          <div className={`grid grid-cols-1 ${showMatchingUsers ? 'lg:grid-cols-3' : ''} gap-8`}>
+            {/* Left Column - Configuration Details */}
+            <div className={`${showMatchingUsers ? 'lg:col-span-2' : ''} space-y-6`}>
+              {/* Shared Configuration */}
+              <Card size="full" className={`${isSimpleView ? 'hidden' : ''}`}>
+                <CardHeader className="pb-4">
+                  <CardTitle data-testid={'share-config-title'} className="text-lg font-medium text-gray-900">
+                    Shared configuration
+                  </CardTitle>
+                </CardHeader>
+                <CardContent className="space-y-4">
+                  <div className="grid grid-cols-2 gap-6">
+                    <div>
+                      <div className="text-sm font-medium text-gray-700 mb-1">Principal</div>
+                      <div className="text-sm text-gray-900">
+                        <span className="text-gray-600">{data.principalType}:</span>{' '}
+                        <span>{data.sharedConfig.principal.split(':')[1]}</span>
```

Comment:
These `split (':')[1]` occur on quite a lot of places. Would it be an idea to create a pure function, add some good naming and documentation so that the code is more readable?

Sth like this (role, name values are arbitrary).

`const {role, name} = parsePrincipal(data.sharedConfig.principal)`

Reply:
good idea